### PR TITLE
Add Peer to serviceDefaults

### PR DESCRIPTION
**Pull request by DanStough**
Changes proposed in this PR:
- Adds a new `Peer` field to serviceDefaults.upstreamConig.Overrides that disambiguates overrides for local vs. peered services. 
- Fixes CRD generation for peeringDialer and peeringAcceptor

How I've tested this PR:
Unit tests.

How I expect reviewers to test this PR:
👁️ 👄 👁️ 

Checklist:
- [X] Tests added
- [ ] CHANGELOG entry added

### CODE REVIEW

File: control-plane/api/v1alpha1/servicedefaults_types.go
Comment (wilkermichael) [inline control-plane/api/v1alpha1/servicedefaults_types.go:380]: Good catch!

**Review by wilkermichael — COMMENTED** [review]
LGTM!

I'm going to tag another former K8s member just to double check

**Review by wilkermichael — APPROVED** [review]

File: control-plane/api/v1alpha1/servicedefaults_types.go
Comment (thisisnotashwin) [inline control-plane/api/v1alpha1/servicedefaults_types.go:411]: I think we need an additional validation check that both Peer and Partition are not provided simultaneously IIRC?

File: hack/copy-crds-to-chart/main.go
Comment (thisisnotashwin) [inline hack/copy-crds-to-chart/main.go:17]: ❤️

**Review by thisisnotashwin — APPROVED** [review]
This is delightful! Left a comment about a missing scenario but I don't need to do another review after that!

File: control-plane/api/v1alpha1/servicedefaults_types.go
Comment (DanStough) [reply control-plane/api/v1alpha1/servicedefaults_types.go:411]: Yep, that's a good callout.

**Comment by DanStough** [comment]
I'm going to hold off merging this until the API changes are in Consul `main`.

**Review by malizz — APPROVED** [review]

**Pull request merged**
